core/services/chainlink/application.go

```diff
@@ -121,6 +124,34 @@ type ChainlinkApplication struct {
 	startStopMu sync.Mutex
 }
 
+func CheckSquashUpgrade(db *gorm.DB) error {
+	// Ensure that we don't try to run a node version later than the
+	// squashed database versions without first migrating up to just before the squash.
+	// If we don't see the latest migration and node version >= S, error and recommend
+	// first running version S - 1 (S = version in which migrations are squashed).
+	if static.Version == "unset" {
+		return nil
+	}
+	squashVersionMinus1 := semver.New("0.9.9")
```

Comment:
may need to bump these values prior to merging, depending on what gets merged in the mean time